config/metrics/patch-metrics-server-for-kind.yml

```diff
@@ -0,0 +1,23 @@
+#@ load("@ytt:data", "data")
+#@ load("@ytt:overlay", "overlay")
+
+#@ if data.values.patch_for_kind:
```

Comment:
I love that this `patch_for_kind` value helps apply multiple yml files - which I think is a big part of why we wanted to move towards this values API/interface.

Reply:
I don't have a strong opinion or deep knowledge here, but we may want to think about separating the automount-service-account-token thing from being 1:1 coupled with using Kind. Kind happens to be a Kubernetes provider that doesn't support [third party tokens](https://istio.io/latest/docs/ops/best-practices/security/#configure-third-party-service-account-tokens) right now, but:

1) There are probably other Kubernetes providers that also don't support this (likely including Minikube, but I haven't checked), and
2) Maybe someday Kind will start supporting this feature, so we wouldn't want to automount service account tokens there anymore, but we do need to apply other sorts of changes specific to Kind.

Just food for thought.

Reply:
Oh, I see now something that might be related to this concern in a comment below :)